ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -170,17 +192,68 @@ public TransactionSimulator(
     return Optional.of(new TransactionSimulatorResult(transaction, result));
   }
 
-  public Optional<Boolean> doesAddressExistAtHead(final Address address) {
-    return doesAddressExist(address, blockchain.getChainHeadHeader());
+  // return combined private/public world state updater if GoQuorum mode, otherwise the public state
+  private WorldUpdater getEffectiveWorldStateUpdater(
+      final BlockHeader header, final MutableWorldState publicWorldState) {
+
+    if (maybePrivacyParameters.isPresent()
+        && maybePrivacyParameters.get().getGoQuorumPrivacyParameters().isPresent()) {
+
+      final MutableWorldState privateWorldState =
+          getPrivateWorldState(
+              maybePrivacyParameters.get().getGoQuorumPrivacyParameters(),
+              header.getStateRoot(),
+              header.getHash());
+      return new GoQuorumMutablePrivateAndPublicWorldStateUpdater(
+          publicWorldState.updater(), privateWorldState.updater());
+    }
+    return publicWorldState.updater();
   }
 
-  public Optional<Boolean> doesAddressExist(final Address address, final BlockHeader header) {
-    if (header == null) {
-      return Optional.empty();
+  private MutableWorldState getPrivateWorldState(
+      final Optional<GoQuorumPrivacyParameters> goQuorumPrivacyParameters,
+      final Hash worldStateRootHash,
```

Comment:
maybe just pass in the header and get the root hash and block hash from the header.